bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/internal/findandreplace/overlay/FindReplaceOverlay.java

```diff
@@ -143,26 +134,46 @@ private final class KeyboardShortcuts {
 	private ToolItem replaceButton;
 	private ToolItem replaceAllButton;
 
-	private Color backgroundToUse;
+	private Color widgetBackgroundColor;
 	private Color normalTextForegroundColor;
 	private boolean positionAtTop = true;
-	private final TargetPartVisibilityHandler targetPartVisibilityHandler;
 	private ContentAssistCommandAdapter contentAssistSearchField, contentAssistReplaceField;
 
+	private class FixColorComposite extends Composite {
```

Comment:
Why use this kind of composite? What do we fix here?

Reply:
It fixes that the CSS engine repeatedly overwrites the background color of composite with one defined by the theme. This is definitely not a perfect way to deal with the necessity to use some other component's background color here, but it's the only workaround I found so far. I am open for a better solution.

Note that this does not mean that the component is not properly themed: it just takes the themed background color of the shell instead of the one of the overlay's outer composite.

Reply:
I think we need to check again how the styling behaves in the overlay in general. Currently, the widgets inherit styling from the outer composite (usually the the composite of the editor's StyledText) except for the outermost composite within the overlay, which would otherwise simply be black in dark theme (i.e., have the same color as the editor, thus without any border). But I am not sure what happens if the composite was something else, i.e., if the colors will still fit then. Note that the proposed solution gets rid of all the existing custom color applications in the current implementation. It only needs to style two widgets instead of all the others.

We should definitely have another look at this topic.